### DISCUSSION TITLE
Dependency management for BaseSettings

### DISCUSSION THREAD

**Discussion by ncadou** [pydantic/pydantic#7368] [Feature Request]
This is a V1 vs V2 question. This is not a question about how to use `BaseSettings`, but about a problem with declaring dependencies.

I am writing a library that uses `BaseSettings`. This being a library, I don't want to force its users into Pydantic V1 or V2. Let the dependent app decide, and my lib will transparently adapt to one or the other.

The issue here is that my lib can either declare `pydantic` as a dependency, which will work for V1 users but not V2, or `pydantic-settings`, which will work with V2 but not V1. AFAICT, both of these options are mutually exclusive, leaving no choice for supporting both.

I can imagine two potential solutions here.

1. Have both Pydantic V1 and V2 declare an extra `pydantic[settings]` that would be a no-op in V1, and my lib could declare that as its dependency.
2. Publish a dummy `pydantic-settings` V1 package that would depend on `pydantic<2`, and my lib would use that (`pydantic-settings`) as its dependency.

Is this something you would consider? Or is there another solution I have missed?

**Comment by sydney-runkle** [comment]
According to the [migration guide](https://docs.pydantic.dev/2.0/migration/#continue-using-pydantic-v1-features), you can continue to use the `Pydantic` V1 API through `pydantic.v1`.

Knowing this, you could set `pydantic` V2 as a dependency, knowing that you're not constraining your users to either version. That being said, we'd recommend migrating to V2 given the significant improvements and new features.
